src/Sentry/Internal/MainSentryEventProcessor.cs

```diff
@@ -80,18 +83,23 @@ public SentryEvent Process(SentryEvent @event)
                 @event.Contexts.Device.Timezone = timeZoneInfo;
             }
 
-            const string currentUiCultureKey = "CurrentUICulture";
-            if (!@event.Contexts.ContainsKey(currentUiCultureKey)
-                && CultureInfoToDictionary(CultureInfo.CurrentUICulture) is { } currentUiCultureMap)
+            IDictionary<string, string> cultureInfoMapped;
+            if (!@event.Contexts.ContainsKey(CultureInfoKey)
+                && CultureInfoToDictionary(CultureInfo.CurrentCulture) is { } currentCultureMap)
             {
-                @event.Contexts[currentUiCultureKey] = currentUiCultureMap;
+                cultureInfoMapped = currentCultureMap;
+                @event.Contexts[CultureInfoKey] = currentCultureMap;
+            }
+            else
+            {
+                cultureInfoMapped = new Dictionary<string, string>();
             }
 
-            const string cultureInfoKey = "CurrentCulture";
-            if (!@event.Contexts.ContainsKey(cultureInfoKey)
-                && CultureInfoToDictionary(CultureInfo.CurrentCulture) is { } currentCultureMap)
+            if (!@event.Contexts.ContainsKey(CurrentUiCultureKey)
+                && CultureInfoToDictionary(CultureInfo.CurrentUICulture) is { } currentUiCultureMap
+                && currentUiCultureMap.Any(p => cultureInfoMapped.Contains(p) is false))
```

Comment:
Just realized there's finally a `TryAdd`:
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.tryadd?view=net-5.0

So we could avoid the double lookup, but would require adding a polifyll for ns2.0 and net4x.

Also, `is false`, maybe stay with `!` instead?
```suggestion
                && !currentUiCultureMap.Any(p => cultureInfoMapped.Contains(p)))
```